lib/web/JsonpChunkTemplatePlugin.js

```diff
@@ -30,6 +31,47 @@ class JsonpChunkTemplatePlugin {
 				const jsonpFunction = chunkTemplate.outputOptions.jsonpFunction;
 				const globalObject = chunkTemplate.outputOptions.globalObject;
 				const source = new ConcatSource();
+				let externals = [];
+
+				if (!chunk.entryModule) {
+					externals = []
+						.concat(
+							...chunk
+								.getModules()
+								.map(m =>
+									m.dependencies.filter(
+										d => d.module && d.module.externalType === "amd"
+									)
+								)
+						)
+						.map(d => d.module);
+
+					externals = externals.filter(function(value, index, self) {
+						return self.indexOf(value) === index && value;
+					});
+				}
+
+				const externalsDepsArray = JSON.stringify(
+					externals.map(m =>
+						typeof m.request === "object" ? m.request.amd : m.request
+					)
+				);
+
+				const externalsArguments = externals
+					.map(
+						m =>
+							`__WEBPACK_EXTERNAL_MODULE_${Template.toIdentifier(
```

Comment:
This string interpolation template is repeated 8 times, you may want to consider creating a method that accepts the request and returns the resulting string.